JSImpl/src/ByteCodeGenerator.cpp

```diff
@@ -0,0 +1,390 @@
+#include "ByteCodeGenerator.h"
+#include "ExpressionVisitor.h"
+#include <algorithm>
+
+class ByteCodeGenerator : public ExpressionVisitor
+{
+public:
+	ByteCodeGenerator() {};
+	~ByteCodeGenerator() {};
+
+
+	virtual void Visit(FunctionDeclaration* e) override;
+	virtual void Visit(BlockStatement* e) override;
+	virtual void Visit(BinaryExpression* e) override;
+	virtual void Visit(LiteralNumber* e) override;
+	virtual void Visit(TopStatements* e) override;
+	virtual void Visit(ListExpression* e) override;
+	virtual void Visit(VariableDefinitionExpression* e) override;
+
+	IPLString GetCode();
+	unsigned ResolveRegisterName(IPLString& name);
+private:
+
+	struct Instruction
+	{
+		enum Type : char
+		{
+			ADD,
+			SUB,
+			MUL,
+			DIV,
+			MOD,
+			MOV,
+			PRINT,
+			READ,
+			CALL,
+			RET,
+			JMP,
+			JMPT,
+			JMPF,
+			DUP,
+			PUSH,
+			POP,
+			SAVE,
+			RESTORE,
+			LESS,
+			EQ,
+			LEQ,
+			GETG,
+			SETG,
+			GETUP,
+			SETUP,
+			GET,
+			SET,
+			INC,
+			DEC,
+			AND,
+			OR,
+			XOR,
+			NOT,
+			THROW,
+			CATCH,
+			CONST,
+			STRING,
+			HALT,
+		};
+
+		Type Descriptor;
+		IPLString Args[3];
+		double Number;
+	};
+	IPLVector<IPLString> m_RegisterTable;
+	IPLVector<Instruction> m_Code;
+
+	IPLStack<IPLString> m_RegisterStack;
+	unsigned m_CurrentStackPointer;

	IPLString m_OutputCode;
};

```

Comment:
This is a coding style guide violation.

Reply:
something weird is happening between my editor and line ending in this file

Reply:
but anyway it's outdated :D

Reply:
Not sure why I was seeing the outdated version.